File: doc/cli/examples/Net/CxUdpClient/CxUdpClient_01.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace User
{
	partial class Examples
	{
		[XIE.CxPluginExecute]
		public void CxUdpClient_01()
		{
			XIE.Net.CxUdpClient client1 = new XIE.Net.CxUdpClient();
			XIE.Net.CxUdpClient client2 = new XIE.Net.CxUdpClient();

			try
			{
				XIE.Net.TxIPEndPoint ep1 = new XIE.Net.TxIPEndPoint(System.Net.IPAddress.Loopback, 5001);
				XIE.Net.TxIPEndPoint ep2 = new XIE.Net.TxIPEndPoint(System.Net.IPAddress.Loopback, 5002);
				int seqno = 0;

				client1.EndPoint = ep1;
				client2.EndPoint = ep2;

				client1.Setup();
				client2.Setup();

				// ----------------------------------------------------------------------
				Console.WriteLine("[{0}] client1.Write()", ++seqno);
				{
					string text = "Hola, mundo!";
					byte[] buf = Encoding.UTF8.GetBytes(text);

					XIE.Net.TxIPEndPoint remoteEP = ep2;

					int count = client1.Write(buf, buf.Length, 1000, remoteEP);
					Console.WriteLine("[{0}] client1.Write returned {1}. [{2}]", seqno, count, text);
				}

				// ----------------------------------------------------------------------
				Console.WriteLine("[{0}] client2.Read()", ++seqno);
				if (client2.Readable(1000))
				{
					byte[] buf = new byte[256];

					XIE.Net.TxIPEndPoint remoteEP = new XIE.Net.TxIPEndPoint(System.Net.IPAddress.Any, 0);

					int count = client2.Read(buf, buf.Length, 1000, ref remoteEP);

					string text = Encoding.UTF8.GetString(buf, 0, count);
					Console.WriteLine("[{0}] client2.Read returned {1}. [{2}] (Port={3})", seqno, count, text, remoteEP.Port);
				}
			}
			finally
			{
				client1.Dispose();
				client2.Dispose();
			}
		}
	}
}

File: doc/cpp/examples/Net/CxUdpClient/CxUdpClient_01.cpp

#include "xie_core.h"
#include "xie_high.h"
#include <stdio.h>

namespace User
{

// ============================================================
void CxUdpClient_01()
{
	printf("%s\n", __FUNCTION__);

	xie::Net::CxUdpClient client1;
	xie::Net::CxUdpClient client2;

	try
	{
		const xie::Net::TxIPEndPoint	ep1(xie::Net::TxIPAddress::Loopback(), 5001);
		const xie::Net::TxIPEndPoint	ep2(xie::Net::TxIPAddress::Loopback(), 5002);
		int seqno = 0;

		client1.EndPoint(ep1);
		client2.EndPoint(ep2);

		client1.Setup();
		client2.Setup();

		// ----------------------------------------------------------------------
		printf("[%d] client1.Write()\n", ++seqno);
		{
			const char*	buf = "Hola, mundo!";
			int			buflen = strlen(buf);

			xie::Net::TxIPEndPoint	remoteEP = ep2;

			int count = client1.Write(buf, buflen, 1000, remoteEP);
			printf("[%d] client1.Write returned %d. [%s]\n", seqno, count, buf);
		}

		// ----------------------------------------------------------------------
		printf("[%d] client2.Read()\n", ++seqno);
		if (client2.Readable(1000))
		{
			const int	buflen = 256;
			char		buf[buflen] = {0};

			xie::Net::TxIPEndPoint	remoteEP(xie::Net::TxIPAddress::Any(), 0);

			int count = client2.Read(buf, buflen, 1000, remoteEP);
			printf("[%d] client2.Read returned %d. [%s] (Port=%d)\n", seqno, count, buf, remoteEP.Port);
		}
	}
	catch(const xie::CxException& ex)
	{
		printf("%s(%d): Code=%d Function=%s\n", ex.File(), ex.Line(), ex.Code(), ex.Function());
	}

	client1.Dispose();
	client2.Dispose();
}

}
